File: src/types/masterData.ts
export interface SystemHealth {
  appName: string
  version: string
  environment: string
  status: string
}

export interface ProductionTab {
  key: string
  label: string
  icon: string
  route: string
}

export interface Unit {
  id: number
  code: string
  name: string
  symbol: string
}

export interface Material {
  id: number
  code: string
  name: string
  stockUnitId: number
  purchaseUnitId: number
  currentCost: number | null
  currentCostUnitId: number | null
  lastCostAt: string | null
}

export interface MaterialCostHistoryEntry {
  id: number
  materialId: number
  unitId: number
  unitCost: number
  effectiveAt: string
}

export interface FinishedProduct {
  id: number
  code: string
  name: string
  unitId: number
  recipeId: number | null
  category: string
  status: string
  internalCode: string | null
  commercialCode: string | null
}

export interface RecipeItem {
  materialId: number
  quantity: number
  unitId: number
}

export interface Recipe {
  id: number
  code: string
  name: string
  productId: number | null
  status: string
  version: string
  outputQuantity: number
  outputUnitId: number
  notes: string | null
  items: RecipeItem[]
}

export interface UnitConversion {
  id: number
  fromUnitId: number
  toUnitId: number
  factor: number
}

export interface MasterDataSnapshot {
  units: Unit[]
  materials: Material[]
  products: FinishedProduct[]
  recipes: Recipe[]
  conversions: UnitConversion[]
}

export interface CreateUnitInput {
  code: string
  name: string
  symbol: string
}

export interface UpdateUnitInput {
  id: number
  code: string
  name: string
  symbol: string
}

export interface CreateMaterialInput {
  code: string
  name: string
  stockUnitId: number
  purchaseUnitId: number
  initialCost?: number | null
  initialCostUnitId?: number | null
}

export interface UpdateMaterialInput {
  id: number
  code: string
  name: string
  stockUnitId: number
  purchaseUnitId: number
}

export interface RecordMaterialCostInput {
  materialId: number
  unitId: number
  unitCost: number
}

export interface CreateFinishedProductInput {
  code: string
  name: string
  unitId: number
  recipeId?: number | null
  category: string
  status: string
  internalCode?: string | null
  commercialCode?: string | null
}

export interface UpdateFinishedProductInput {
  id: number
  code: string
  name: string
  unitId: number
  recipeId?: number | null
  category: string
  status: string
  internalCode?: string | null
  commercialCode?: string | null
}

export interface CreateRecipeInput {
  code: string
  name: string
  productId?: number | null
  status: string
  version: string
  outputQuantity: number
  outputUnitId: number
  notes?: string | null
  items: RecipeItem[]
}

export interface UpdateRecipeInput {
  id: number
  code: string
  name: string
  productId?: number | null
  status: string
  version: string
  outputQuantity: number
  outputUnitId: number
  notes?: string | null
  items: RecipeItem[]
}

export interface CreateUnitConversionInput {
  fromUnitId: number
  toUnitId: number
  factor: number
}

export interface UpdateUnitConversionInput {
  id: number
  fromUnitId: number
  toUnitId: number
  factor: number
}

export interface ConvertUnitsInput {
  fromUnitId: number
  toUnitId: number
  quantity: number
}

export interface ConversionResult {
  factor: number
  convertedQuantity: number
}

File: src-tauri/src/domain/master_data.rs
#[derive(Clone, Debug, serde::Serialize, sqlx::FromRow)]
#[serde(rename_all = "camelCase")]
pub struct Unit {
    pub id: i64,
    pub code: String,
    pub name: String,
    pub symbol: String,
}

#[derive(Clone, Debug, serde::Serialize, sqlx::FromRow)]
#[serde(rename_all = "camelCase")]
pub struct Material {
    pub id: i64,
    pub code: String,
    pub name: String,
    pub stock_unit_id: i64,
    pub purchase_unit_id: i64,
    pub current_cost: Option<f64>,
    pub current_cost_unit_id: Option<i64>,
    pub last_cost_at: Option<String>,
}

#[derive(Clone, Debug, serde::Serialize, sqlx::FromRow)]
#[serde(rename_all = "camelCase")]
pub struct MaterialCostHistoryEntry {
    pub id: i64,
    pub material_id: i64,
    pub unit_id: i64,
    pub unit_cost: f64,
    pub effective_at: String,
}

#[derive(Clone, Debug, serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FinishedProduct {
    pub id: i64,
    pub code: String,
    pub name: String,
    pub unit_id: i64,
    pub recipe_id: Option<i64>,
    pub category: String,
    pub status: String,
    pub internal_code: Option<String>,
    pub commercial_code: Option<String>,
}

#[derive(Clone, Debug, serde::Serialize, sqlx::FromRow)]
#[serde(rename_all = "camelCase")]
pub struct RecipeItem {
    pub recipe_id: i64,
    pub material_id: i64,
    pub quantity: f64,
    pub unit_id: i64,
}

#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RecipeItemPayload {
    pub material_id: i64,
    pub quantity: f64,
    pub unit_id: i64,
}

#[derive(Clone, Debug, serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Recipe {
    pub id: i64,
    pub code: String,
    pub name: String,
    pub product_id: Option<i64>,
    pub status: String,
    pub version: String,
    pub output_quantity: f64,
    pub output_unit_id: i64,
    pub notes: Option<String>,
    pub items: Vec<RecipeItemPayload>,
}

#[derive(Clone, Debug, serde::Serialize, sqlx::FromRow)]
#[serde(rename_all = "camelCase")]
pub struct UnitConversion {
    pub id: i64,
    pub from_unit_id: i64,
    pub to_unit_id: i64,
    pub factor: f64,
}

#[derive(Clone, Debug, serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct MasterDataSnapshot {
    pub units: Vec<Unit>,
    pub materials: Vec<Material>,
    pub products: Vec<FinishedProduct>,
    pub recipes: Vec<Recipe>,
    pub conversions: Vec<UnitConversion>,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateUnitInput {
    pub code: String,
    pub name: String,
    pub symbol: String,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateUnitInput {
    pub id: i64,
    pub code: String,
    pub name: String,
    pub symbol: String,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateMaterialInput {
    pub code: String,
    pub name: String,
    pub stock_unit_id: i64,
    pub purchase_unit_id: i64,
    pub initial_cost: Option<f64>,
    pub initial_cost_unit_id: Option<i64>,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateMaterialInput {
    pub id: i64,
    pub code: String,
    pub name: String,
    pub stock_unit_id: i64,
    pub purchase_unit_id: i64,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RecordMaterialCostInput {
    pub material_id: i64,
    pub unit_id: i64,
    pub unit_cost: f64,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateFinishedProductInput {
    pub code: String,
    pub name: String,
    pub unit_id: i64,
    pub recipe_id: Option<i64>,
    pub category: String,
    pub status: String,
    pub internal_code: Option<String>,
    pub commercial_code: Option<String>,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateFinishedProductInput {
    pub id: i64,
    pub code: String,
    pub name: String,
    pub unit_id: i64,
    pub recipe_id: Option<i64>,
    pub category: String,
    pub status: String,
    pub internal_code: Option<String>,
    pub commercial_code: Option<String>,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateRecipeInput {
    pub code: String,
    pub name: String,
    pub product_id: Option<i64>,
    pub status: String,
    pub version: String,
    pub output_quantity: f64,
    pub output_unit_id: i64,
    pub notes: Option<String>,
    pub items: Vec<RecipeItemPayload>,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateRecipeInput {
    pub id: i64,
    pub code: String,
    pub name: String,
    pub product_id: Option<i64>,
    pub status: String,
    pub version: String,
    pub output_quantity: f64,
    pub output_unit_id: i64,
    pub notes: Option<String>,
    pub items: Vec<RecipeItemPayload>,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateUnitConversionInput {
    pub from_unit_id: i64,
    pub to_unit_id: i64,
    pub factor: f64,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateUnitConversionInput {
    pub id: i64,
    pub from_unit_id: i64,
    pub to_unit_id: i64,
    pub factor: f64,
}

#[derive(Debug, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ConvertUnitsInput {
    pub from_unit_id: i64,
    pub to_unit_id: i64,
    pub quantity: f64,
}

#[derive(Debug, serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ConversionResult {
    pub factor: f64,
    pub converted_quantity: f64,
}
